qiskit/algorithms/amplitude_estimators/ae.py

```diff
@@ -169,7 +170,7 @@ def evaluate_measurements(
                 y measurements with respective probabilities, in this order.
         """
         # compute grid sample and measurement dicts
-        if isinstance(circuit_results, dict):
+        if isinstance(circuit_results, (dict, Mapping)):
```

Comment:
```suggestion
        if isinstance(circuit_results, Mapping):
```
because goose types as a mapping (and other instances below).